src/LinuxCaptureService/MemoryWatchdog.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef LINUX_CAPTURE_SERVICE_MEMORY_WATCHDOG_H_
+#define LINUX_CAPTURE_SERVICE_MEMORY_WATCHDOG_H_
+
+#include <optional>
+#include <string_view>
+#include <thread>
+
+namespace orbit_linux_capture_service {
+
+// In header file for testing.
```

Comment:
If you consider this internal API, how about putting this into an _internal namespace?

Reply:
We decided to use `_internal` for classes/functions that need to be in the public headers of the module but shouldn't be used by other modules. This is in line with `_internal` suffixing the namespace of the module. But this is a private header, so that doesn't apply&mdash;after all, all its content is already internal to the module.

Reply:
Ah yeah. You're right. I completely missed that it's a private header.